main.rb

```diff
@@ -31,6 +22,50 @@ def list_people
     puts
   end
 
+  def list_rentals_for_person
+    print 'ID of person: '
+    id_of_person = gets.chomp.to_i
+    puts 'Rentals:'
+    @rentals.each do |rent|
+      puts "Date: #{rent.date}, Book #{rent.book.title} by #{rent.book.author}" if rent.person.id == id_of_person
+    end
+    puts
+  end
+end
+
+class CreateClass
+  def initialize(list_handler)
+    @list_handler = list_handler
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    b = Book.new(title, author)
+    @list_handler.books.push(b)
+    puts 'Book created successfully'
+    puts
+  end
+
+  def create_rental
+    puts 'Select a book from the following list'
+    @list_handler.books.each_with_index { |b, idx| puts "[#{idx}] Title: #{b.title}, Author: #{b.author}" }
+    book_idx = gets.chomp.to_i
+    puts 'Select a person from the following list'
+    @list_handler.people.each_with_index do |person, idx|
+      puts "[#{idx}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_idx = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+    rental = Rental.new(date, @list_handler.books[book_idx], @list_handler.people[person_idx])
+    @list_handler.rentals.push(rental)
+    puts 'Rental created successfully'
+    puts
+  end
+
```

Comment:
- [ ] great job on cleaning up the main.rb file 👍 however the `create_student` and `create_teacher` method is not in line with the SOLID principle in particular the single responsibility rule. please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) provided by Microverse for more information.
- [ ] also these methods are not in line with the DRY method, part of the code can be re-used